sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/init.sql

```diff
@@ -204,7 +213,7 @@ SELECT
   DATE(first_seen.first_run_datetime) AS first_run_date,
   first_seen.first_reported_country AS first_reported_country,
   first_seen.first_reported_isp AS first_reported_isp,
-  first_seen.channel AS channel,
+  COALESCE(first_seen.channel, first_session.channel, metrics.channel) AS channel,
```

Comment:
That's how I think is best handled.

In theory, since the source is the same and [there are only 2 clients without first_seen_date](https://sql.telemetry.mozilla.org/queries/94111/source), the null channel shouldn't happen after the initialization. We could validate this in analysis or with a more rigorous test?

-- Another option I considered is to use
COALESCE(first_seen.channel, first_session.channel, metrics.channel, baseline.channel) AS channel,
to always identify the channel. But this would not be according to the logic of the funnel analysis for which we require the differentiation between first and last reported values.

What do you think works best?

Reply:
My idea is that If the client's channel is NULL here, then the `last_reported_channel` would enable identify from which channel they're being retrieved and they would be expected to have nulls for all the reported ping columns in the metadata.